include/utils/message_queue.hpp

```diff
@@ -68,7 +74,7 @@ class MessageHandler {
     ~MessageHandler();
 
     /// \brief Adds a \p message to the message queue which will be delivered to the registered handlers
-    void add(std::shared_ptr<json> message);
+    void add(MessageDetails message);
```

Comment:
Passing by value here is alright, but then, for efficiency, the caller should also move it in

Reply:
The recommendation comes from clang-tidy suggestion (for constructors)
```
MyClass(std::string s) : m_string{std::move(s)} {}
```
Where the caller doesn't use move.
An alternative is
```
MyClass(std::string &&s) : m_string{std::move(s)} {}
```
where move would be required

In this particular case add is called:
```
handler.add({topic, data});

handler.add(std::move({topic, data})); // this gives a compile error
```

Reply:
Yes, that suggestion is correct.  The caller doesn't need to move it in, but has the flexibility to do so (and in our case we should do so, because we probably won't need the topic and data anymore afterwards).
If you call the method with aggregate initialization, the `{topic, data}` is already a temporary and doesn't need to be moved.  `std::move({topic, data})` probably gives a compiler error, because it can't deduce the type, that should be aggregated?